sharding/nodesSetup.go

```diff
@@ -118,9 +144,64 @@ func NewNodesSetup(
 	nodes.processShardAssignment()
 	nodes.createInitialNodesInfo()
 
+	epochNotifier.RegisterNotifyHandler(nodes)
+
 	return nodes, nil
 }
 
+// EpochConfirmed is called whenever a new epoch is confirmed
+func (ns *NodesSetup) EpochConfirmed(epoch uint32, _ uint64) {
+	ns.mutConfiguration.Lock()
+	defer ns.mutConfiguration.Unlock()
+
+	if ns.currentChainParameters.EnableEpoch == epoch {
```

Comment:
this `if` is somehow useless. EpochConfirmed is called once per epoch

Reply:
refactored